spec/support/acceptance/matchers/match_without_whitespace.rb

```diff
@@ -0,0 +1,9 @@
+RSpec::Matchers.define :match_without_whitespace do |expected|
+  match do |actual|
+    actual.gsub(/\s*/, '').match?(Regexp.new(expected.source, Regexp::EXTENDED))
+  end
+
+  failure_message do |actual|
+    "Actual:\n\n\s\s#{actual.gsub(/\s*/, '')}\n\nExpected:\n\n\s\s#{expected.source}"
```

Comment:
This is a little confusing because it says "actual" but really it's the actual value after the whitespace has been removed.

Which do you think would be more useful to see for the developer debugging a test failure?

Reply:
I think Actual is fine as that is the standard format. I took this for a spin and ensured a failure. My trouble comes with how to parse the output as its a bit hard to see:

```
  1) certs with default params Command "certutil -L -d /etc/pki/katello/nssdb -n ca" stdout is expected to match without whitespace /Issuer: "CN=centos7-64.example.com,OU=SomeOrgUni,O=Katello,L=Raleigh,ST=North Carolina,C=US"/
     Failure/Error: its(:stdout) { should match_without_whitespace(/Issuer: "CN=#{host_inventory['fqdn']},OU=SomeOrgUni,O=Katello,L=Raleigh,ST=North Carolina,C=US"/) }
       Actual:
       
         Certificate:Data:Version:3(0x2)SerialNumber:00:a4:65:a6:8b:e4:9b:82:a2SignatureAlgorithm:PKCS#1SHA-256WithRSAEncryptionIssuer:"CN=centos7-64.example.com,OU=SomeOrgUnit,O=Katello,L=Raleigh,ST=NorthCarolina,C=US"Validity:NotBefore:FriJun0412:15:562021NotAfter:MonJan1812:15:562038Subject:"CN=centos7-64.example.com,OU=SomeOrgUnit,O=Katello,L=Raleigh,ST=NorthCarolina,C=US"SubjectPublicKeyInfo:PublicKeyAlgorithm:PKCS#1RSAEncryptionRSAPublicKey:Modulus:ad:78:a2:9e:c5:67:ec:f9:4d:82:43:4f:26:be:60:d1:ca:58:f8:9a:8e:c7:05:88:a2:31:b7:4b:50:6c:3f:fe:66:bd:4e:51:43:35:45:ce:ce:fa:98:1d:df:c5:8d:10:2a:20:c0:ef:47:52:a5:8b:ef:eb:dc:94:8d:af:c3:83:b8:ae:02:9a:87:04:ee:5d:3a:cc:c4:91:db:af:3b:2f:fa:ba:9d:3f:f4:6a:0f:01:9a:a9:83:4d:19:73:d9:a5:e5:8a:72:60:08:05:0f:db:3c:19:20:3e:ae:f8:f6:0f:ee:12:91:b2:4f:8e:d2:8d:98:6f:65:e1:32:52:b8:fd:54:c0:12:58:0a:d5:81:47:92:6e:ce:88:05:d6:ac:88:a9:10:cd:0d:97:20:b8:4c:66:02:7e:fc:56:41:39:32:3f:41:a1:fb:ac:5d:62:1f:f0:c6:e1:fa:42:42:a3:78:f8:37:d8:8f:a7:2d:a0:fc:ea:fa:d6:65:bb:a5:d5:29:be:40:16:44:33:6f:c6:82:0a:ce:3c:a9:4b:db:71:a4:70:c3:1a:4c:63:55:6b:fe:e0:1c:7c:f9:ba:ca:b5:78:16:71:b7:96:0e:2d:3e:6b:67:1d:88:59:19:45:c9:8a:96:89:00:8c:ea:b3:8c:45:c9:45:d0:dd:28:46:ef:5f:04:89:56:95:7b:96:a8:03:45:2c:21:5c:2f:87:39:26:06:9f:7a:65:a8:b8:b6:ec:5a:71:c3:91:d1:78:84:e1:9e:f1:6c:7f:33:e8:4c:c8:a8:e3:3d:a3:77:cc:a5:d6:a5:9b:7a:11:64:db:44:27:d4:0f:6a:64:ee:3f:59:d0:90:bb:ec:2d:c9:df:12:03:b3:71:51:42:2b:37:ed:87:23:2e:77:42:2b:32:e7:a9:37:c4:09:44:61:58:19:f1:31:20:e4:d7:9e:5a:10:0c:d9:50:5b:30:eb:16:6b:b0:7c:03:73:ce:5d:26:d5:70:e8:d9:5d:60:0e:4d:7f:4f:85:92:cc:7c:4b:ae:11:8f:8e:21:65:b1:5e:a7:1b:15:21:d8:44:e5:d0:2d:60:e3:3f:25:a8:53:3b:e6:6d:d0:c7:04:14:ba:3d:18:dc:63:fd:eb:b0:65:24:b7:01:8e:2b:7a:d7:17:ba:fa:e6:87:e1:40:5f:bb:6e:90:6e:d5:a8:20:61:69:82:05:1f:02:bf:6d:9e:24:a9:dc:43:52:39:31:7b:e9:bc:67:f7:21:93:6f:df:06:c6:22:72:a1:74:99:5e:b9:58:2c:84:8a:05:09:ef:3d:a2:ed:d9:59:a2:96:20:e3:78:f5:6b:59:e4:5f:dd:28:7e:2b:ed:9bExponent:65537(0x10001)SignedExtensions:Name:CertificateBasicConstraintsData:IsaCAwithnomaximumpathlength.Name:CertificateKeyUsageUsages:DigitalSignatureKeyEnciphermentCertificateSigningCRLSigningName:ExtendedKeyUsageTLSWebServerAuthenticationCertificateTLSWebClientAuthenticationCertificateName:CertificateTypeData:<SSLServer,SSLCA>Name:CertificateCommentComment:"KatelloSSLToolGeneratedCertificate"Name:CertificateSubjectKeyIDData:e6:1a:ce:b0:a7:16:55:03:48:73:13:9b:89:52:82:1c:7a:a9:dd:a3Name:CertificateAuthorityKeyIdentifierKeyID:e6:1a:ce:b0:a7:16:55:03:48:73:13:9b:89:52:82:1c:7a:a9:dd:a3Issuer:DirectoryName:"CN=centos7-64.example.com,OU=SomeOrgUnit,O=Katello,L=Raleigh,ST=NorthCarolina,C=US"SerialNumber:00:a4:65:a6:8b:e4:9b:82:a2SignatureAlgorithm:PKCS#1SHA-256WithRSAEncryptionSignature:59:80:c2:bd:1b:4d:a6:0f:a3:52:a8:8a:52:65:32:50:69:8b:51:29:99:bc:79:64:43:9c:80:df:78:ac:74:5c:c2:f3:c0:d6:fc:6d:19:f0:c1:d7:77:06:66:88:a2:03:58:0b:e1:67:28:03:66:c6:04:e6:b6:84:5b:f7:7d:03:81:76:2e:e6:ba:64:41:fa:0a:16:ad:20:66:b6:06:8c:67:a6:c0:dc:87:cd:de:6a:11:18:5e:c2:ef:6d:39:06:c0:ae:14:12:05:c4:e0:02:8a:06:13:47:af:56:ca:36:ba:f2:75:5e:96:1b:6d:5b:10:cf:cc:c9:68:bb:96:4f:0d:aa:8c:cc:3f:f0:e1:ba:a5:c6:64:c2:e8:f7:8a:2b:20:f4:72:5a:32:64:11:28:47:d2:d8:c8:7d:3b:1f:d6:18:86:f7:03:c0:03:82:da:db:f3:a7:af:cc:33:a5:6a:ba:d1:5b:fe:ef:de:47:8b:29:2a:bb:9a:84:44:d2:2b:7f:cf:b4:a5:81:75:e5:62:3e:68:18:b6:42:a2:a0:72:93:a0:91:7a:6c:51:ec:36:5b:4d:a7:96:7a:75:06:ba:68:ca:27:fb:bc:7d:27:63:73:58:70:f9:ef:b1:3a:e6:23:0d:00:30:a4:d9:3b:bf:99:f2:ab:3f:05:49:a8:35:78:f0:60:6b:eb:4a:a4:0a:a8:85:8d:d2:aa:4d:41:c1:ee:54:ce:17:89:74:00:a3:a5:9b:dd:e8:63:c9:58:f2:db:66:fb:44:82:32:8d:72:cc:31:31:df:9c:35:0b:79:3c:10:96:ec:89:98:71:42:24:a7:8d:22:da:9c:8b:1a:e9:2a:49:b4:27:e0:e3:cf:b2:78:c2:bb:ba:87:4d:39:52:85:80:40:8a:e8:ba:a9:46:66:ff:b1:1a:69:98:83:29:9c:fa:be:02:2f:cf:33:2c:b3:51:0d:19:75:73:28:5f:74:27:a5:f5:7f:87:ae:39:de:b6:be:35:63:01:87:6b:9c:2c:9e:7b:c3:3a:ca:7f:8d:bb:28:8a:e5:2e:ac:7a:9e:98:b8:0c:13:83:7b:24:bf:a1:9d:12:10:ea:34:1e:72:d7:f3:d2:85:7d:e1:ce:b7:32:30:a1:e2:79:36:b2:9b:ee:3a:c9:c5:e9:70:df:94:e0:d4:7a:f5:81:76:ba:0e:e8:0d:50:e8:48:54:51:d3:98:c7:86:b8:38:07:88:1d:d3:2b:ca:b3:f6:1e:5a:97:51:f5:3f:f3:4b:81:2a:e1:31:72:ae:07:b2:37:29:7c:fa:d6:f3:17:c5:a3:83:a7:13:bf:b5:09:2f:aa:fe:2c:3a:59:66:cd:94:76Fingerprint(SHA-256):2E:DA:42:12:89:0A:45:50:75:07:7D:06:54:C8:23:12:64:12:F3:71:42:0C:8D:89:5C:52:56:55:1E:B0:01:DEFingerprint(SHA1):46:91:45:E9:A9:76:9D:C8:7F:BC:9C:8C:84:FD:DD:22:55:70:86:19Mozilla-CA-Policy:false(attributemissing)CertificateTrustFlags:SSLFlags:ValidCATrustedCATrustedClientCAEmailFlags:ValidCATrustedCAObjectSigningFlags:ValidCA
       
       Expected:
       
         Issuer: "CN=centos7-64.example.com,OU=SomeOrgUni,O=Katello,L=Raleigh,ST=North Carolina,C=US"
```

If you compare to a regular match, it tends to have an indicator of the "diff":

```
     Failure/Error: its(:stdout) { should match(/^broke                                                       u,u,u$/i) }
       expected "\nCertificate Nickname                                         Trust Attributes\n                   ...                       CT,C,c\nbroker                                                       u,u,u\n" to match /^broke                                                       u,u,u$/i
       Diff:
       @@ -1,6 +1,11 @@
       -/^broke                                                       u,u,u$/i
       +
       +Certificate Nickname                                         Trust Attributes
       +                                                             SSL,S/MIME,JAR/XPI
       +
       +ca                                                           CT,C,c
       +broker                                                       u,u,u
```

If you were able to somehow incorporate that I think it would help.